assignment 6/promisesassignment.js

```diff
@@ -0,0 +1,53 @@
+var fs = require("fs");
+
+var firstMethod = function() {
+  var promise = new Promise(function(resolve, reject) {
+    fs.readFile("file1.txt", function(err, data) {
+      if (err) {
+        reject(err);
+      }
+      console.log("first method completed");
+      resolve(data);
+    });
+  });
+  return promise;
+};
+
+var secondMethod = function(data) {
+  var promise = new Promise(function(resolve, reject) {
+    fs.appendFile("file2.txt", data, function(err) {
+      if (err) {
+        reject(err);
+      }
+      resolve(console.log("second method completed"));
+    });
+  });
+  return promise;
+};
+
+var thirdMethod = function() {
+  var promise = new Promise(function(resolve, reject) {
+    fs.readFile("file2.txt", function(err, data) {
+      if (err) {
+        reject(err);
+      }
+      console.log("third method completed");
+      resolve(data);
+    });
+  });
+  return promise;
+};
+
+var fourthMethod = function(data) {
+  var promise = new Promise(function(resolve, reject) {
+    fs.appendFile("file3.txt", data, function(err) {
+      if (err) {
+        reject(err);
+      }
+      resolve(console.log("fourth method completed"));
+    });
+  });
+  return promise;
+};
+
+firstMethod().then(secondMethod).then(thirdMethod).then(fourthMethod);
```

Comment:
@abansal26 Catch block is missing here and please use 'then' in new line.

Reply:
Changes as suggested by @sakodiya commited. Commit ID - de7738060960035e58c09b39005e3c5b761817c6

Reply:
@abansal26 'ReferenceError: promise is not defined'. Please check your  program once before creating the PR.